### ISSUE TITLE
Bug Report // Gradle GitHub Dependency Graph Support Action

### ISSUE THREAD

**Issue by camvogel** [gradle/actions#91]
Submission from [GetUpside](https://github.lightning.force.com/lightning/r/Account/0013o00002ZKELSAA5/view) on 3/26/2024

TL;DR: Customer is trying to get Kotlin/Gradle dependencies scanned by Dependabot on one of their critical Android app repositories. Based on our [docs](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#gradle), they have been using this [Action](https://github.com/gradle/actions/tree/main/setup-gradle#github-dependency-graph-support). This is flagging many false positives.

# Summary
The findings are mostly false positives, either a library is flagged by Dependabot that isn't in the project (transitive or not), or it's getting the versions wrong so flagging us for a vulnerable version when we're already using a fixed one. 

We suggested [this](https://github.com/marketplace/actions/build-with-gradle#the-dependency-submission-action), but the customer found it uses the same Action on the backend.

The customer thinks the issue is ultimately with Kotlin+Gradle, and the way versions are defined. They don't have dependency versions defined in their build.gradle.kts file, where it seems like Dependabot is looking for them. So it's able to catch a few of the dependencies they're using but also with a lot of the false positives and incorrect versions mentioned earlier. 

> As I mentioned we're not defining dependency versions in the usual place (build.gradle.kts where it [seems like Dependabot is looking](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#gradle)) so that may be confusing things version-wise. 
> The other issue I've seen before with SCA tools in general is matching dependencies against a given CVE's CPEs. For example, Dependabot is flagging [CVE-2019-20444](https://nvd.nist.gov/vuln/detail/CVE-2019-20444#vulnConfigurationsArea) / [GHSA-cqqj-4p63-rrmm](https://github.com/advisories/GHSA-cqqj-4p63-rrmm) for Netty, but we're not using that at all (transitive or direct). Dependabot is saying it found this via Maven in our settings.gradle.kts:
> > Opened 2 months ago on io.netty:netty-codec-http (Maven) · settings.gradle.kts
>
> But we don't have Netty declared there or pulled in by anything else from what I can tell. I guess my main question is, how is Dependabot matching CVEs? Is it using the NVD CPEs or something you all have built yourselves with your Advisory Database? The tool is a little opaque on that front. 
> Also happy to open an issue on GitHub once I collect some more info on our end.

**Customer Contact:** Charlie Williams (Staff Application Security Engineer) // charles.williams@upside.com

**Comment by bigdaz** [comment]
The `dependency-submission` action resolves all of the dependencies in your build. This includes plugins, dependencies you've declared, test dependencies, and all transitive dependencies of these. It doesn't matter how the dependencies are declared: the ones being _resolved_ by Gradle are the ones being reported.

Many people are surprised to see what dependencies are actually being resolved when they run their builds, but I'm yet to see a case where the dependencies being reported are actually incorrect. If you can confirm that a dependency is really being reported and not resolved by your project, then I can investigate further.

Please [follow the instructions here](https://github.com/gradle/actions/blob/main/dependency-submission/README.md#finding-the-source-of-a-dependency-vulnerability) to identify the source of the dependency version that is being reported. 

Once you have worked out why it is being resolved, you can either [update the dependency version](https://github.com/gradle/actions/blob/main/dependency-submission/README.md#updating-the-dependency-version) or [exclude it from the submitted dependency graph](https://github.com/gradle/actions/blob/main/dependency-submission/README.md#limiting-the-dependencies-that-appear-in-the-dependency-graph).

**Comment by bigdaz** [comment]
You might be interested in reading the (under development) FAQ here: https://github.com/gradle/actions/issues/85

**Comment by camvogel** [comment]
@bigdaz who is the owner of the Gradle org? I couldn't find someone under `People`

This issue needs to be deleted, please.
